xpublish/utils/api.py

```diff
@@ -72,3 +73,50 @@ def check_route_conflicts(routers):
 
     if len(duplicates):
         raise ValueError(f'Found multiple routes defined for the following paths: {duplicates}')
+
+
+class SingleDatasetOpenAPIOverrider:
+    """Used to override the FastAPI application openapi specs when a single
+    dataset is published.
+
+    In this case, the "dataset_id" path parameter is not present in API
+    endpoints and has to be removed manually.
+
+    See:
+
+    - https://fastapi.tiangolo.com/advanced/extending-openapi/
+    - https://github.com/tiangolo/fastapi/issues/1594
+
+    """
+
+    def __init__(self, app):
+        self._app = app
+
+    def openapi(self):
+
+        if self._app.openapi_schema:
+            return self._app.openapi_schema
```

Comment:
@benbovy - looking at the test coverage, this line seems to be missing a test: https://codecov.io/gh/xarray-contrib/xpublish/pull/44/src/xpublish/utils/api.py?before=xpublish/utils/api.py

You should be able to exercise this path by calling `SingleDatasetOpenAPIOverrider(app).openapi()` twice, yeah?